src/vfd/main.c

```diff
@@ -1318,7 +1319,24 @@ main(int argc, char **argv)
 
 		// read PCI config to get VM offset and stride
 		struct rte_eth_dev *pf_dev = &rte_eth_devices[0];
-		rte_eal_pci_read_config(pf_dev->pci_dev, &pci_control_r, 32, 0x174);
+
+		// Find the SR-IOV extended capability structure
+		do {
+			rte_eal_pci_read_config(pf_dev->pci_dev, &pci_control_r, 32, cfg_offset);
+			bleat_printf(4, "Header: %08x (%04x)", pci_control_r, cfg_offset);
+			if ((pci_control_r & 0xffff) == 0x0010)
+				break;
+			cfg_offset = pci_control_r >> 20;
```

Comment:
If it needed masking, it would indicate an error in the PCI configuration I think.